lib/Net/Async/Blockchain/ETH.pm

```diff
@@ -227,13 +256,14 @@ sub subscribe {
                 return undef unless $response->{params} && $response->{params}->{subscription};
                 return $response->{params}->{subscription} eq $self->subscription_id;
             }
-            )->map(
+        )->map(
             async sub {
                 await $self->$subscription(shift);
             }
-            )->ordered_futures->completed(),
+        )->ordered_futures->completed(),
         $self->recursive_search(),
-        )->on_fail(
+        $self->_transform_unprocessed_transactions(),
```

Comment:
As discussed, this will run one time in the beginning, you need a while loop to keep processing the transactions.

Reply:
Added while loop :+1: